service/kas/access/rewrap_test.go

```diff
@@ -394,93 +377,3 @@ func TestParseAndVerifyRequest(t *testing.T) {
 		})
 	}
 }
-
-func TestLegacyBearerTokenFails(t *testing.T) {
-	var tests = []struct {
-		name     string
-		metadata []string
-		msg      string
-	}{
-		{"no auth header", []string{}, "no auth token"},
-		{"multiple auth", []string{"Authorization", "a", "Authorization", "b"}, "auth fail"},
-		{"no bearer", []string{"Authorization", "a"}, "auth fail"},
-		{"no token", []string{"Authorization", "Bearer "}, "auth fail"},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			ctx := metadata.NewIncomingContext(context.Background(), metadata.Pairs(tt.metadata...))
-			p, err := legacyBearerToken(ctx, "")
-			assert.Empty(t, p)
-			require.Error(t, err)
-			assert.Contains(t, err.Error(), tt.msg)
-		})
-	}
-}
-
-func TestLegacyBearerTokenEtc(t *testing.T) {
-	p, err := legacyBearerToken(context.Background(), "")
-	assert.Empty(t, p)
-	require.Error(t, err)
-	assert.Contains(t, err.Error(), "no auth token")
-
-	p, err = legacyBearerToken(context.Background(), "something")
-	assert.Equal(t, "something", p)
-	require.NoError(t, err)
-
-	ctx := metadata.NewIncomingContext(context.Background(), metadata.Pairs("Authorization", "Bearer TOKEN"))
-	p, err = legacyBearerToken(ctx, "")
-	assert.Equal(t, "TOKEN", p)
-	require.NoError(t, err)
-}
-
-func TestHandlerAuthFailure0(t *testing.T) {
-	hsmSession, _ := security.New(&security.HSMConfig{})
-	kasURI, _ := url.Parse("https://" + hostname + ":5000")
-	kas := Provider{
-		URI:            *kasURI,
-		CryptoProvider: hsmSession,
-		OIDCVerifier:   nil,
-	}
-
-	body := `{"mock": "value"}`
-	_, err := kas.Rewrap(context.Background(), &kaspb.RewrapRequest{SignedRequestToken: body})
-	status, ok := status.FromError(err)
-	assert.True(t, ok)
-	assert.Equal(t, codes.Unauthenticated, status.Code())
-}
-
-func TestHandlerAuthFailure1(t *testing.T) {
-	hsmSession, _ := security.New(&security.HSMConfig{})
-	kasURI, _ := url.Parse("https://" + hostname + ":5000")
-	kas := Provider{
-		URI:            *kasURI,
-		CryptoProvider: hsmSession,
-		OIDCVerifier:   nil,
-	}
-
-	body := `{"mock": "value"}`
-	md := map[string][]string{
-		"Authorization": {"Bearer invalidToken"},
-	}
-	ctx := metadata.NewIncomingContext(context.Background(), md)
-	_, err := kas.Rewrap(ctx, &kaspb.RewrapRequest{SignedRequestToken: body})
-	status, ok := status.FromError(err)
-	assert.True(t, ok)
-	assert.Equal(t, codes.PermissionDenied, status.Code())
-}
-
-func TestHandlerAuthFailure2(t *testing.T) {
-	hsmSession, _ := security.New(&security.HSMConfig{})
-	kasURI, _ := url.Parse("https://" + hostname + ":5000")
-	kas := Provider{
-		URI:            *kasURI,
-		CryptoProvider: hsmSession,
-		OIDCVerifier:   nil,
-	}
-
-	body := `{"mock": "value"}`
-	_, err := kas.Rewrap(context.Background(), &kaspb.RewrapRequest{SignedRequestToken: body, Bearer: "invalidToken"})
-	status, ok := status.FromError(err)
-	assert.True(t, ok)
-	assert.Equal(t, codes.PermissionDenied, status.Code())
```

Comment:
I don't see any tests of returning PermissionDenied elsewhere? Can we move these tests to `authn_test.go`?

Reply:
I think permission denied makes sense here but we probably want something to fail if the signed body request is signed with the wrong key. Let me add a few more tests.